File: 2st/Riyeon/리스트 자르기.java
import java.util.*;
class Solution {
    public int[] solution(int n, int[] slicer, int[] num_list) {
        List<Integer> li = new ArrayList<>();
        for(int i=0; i < num_list.length; i++){
            if(n!=1 && i==0)i=slicer[0];
            
            if(n!=2 && i<slicer[1]+1){
                li.add(num_list[i]);
            }else if(n==2)
                li.add(num_list[i]);
            else break;
            
            if(n==4) i+=slicer[2]-1;
        }
        return li.stream().mapToInt(i->i).toArray();
    }
}

File: 1st/Jinseon/리스트자르기.py
def solution(n, slicer, num_list):
    a,b,c=slicer
    if n==1: return num_list[:b+1]
    if n==2: return num_list[a:]
    if n==3: return num_list[a:b+1]
    return num_list[a:b+1:c]
